# VEDA AI & ML Internship – Task 16
## Load a CSV Dataset

This task demonstrates how to load a CSV dataset using Pandas and display its first and last records.

In [4]:
import pandas as pd

In [5]:
url = "https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data"

df = pd.read_csv(
    url,
    header=None,
    names=[
        "SepalLengthCm",
        "SepalWidthCm",
        "PetalLengthCm",
        "PetalWidthCm",
        "Species"
    ]
)

print("CSV dataset loaded successfully!")

CSV dataset loaded successfully!


In [6]:
df.head()

,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
0,5.1,3.5,1.4,0.2,Iris-setosa
1,4.9,3.0,1.4,0.2,Iris-setosa
2,4.7,3.2,1.3,0.2,Iris-setosa
3,4.6,3.1,1.5,0.2,Iris-setosa
4,5.0,3.6,1.4,0.2,Iris-setosa


In [7]:
df.tail()

,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
145,6.7,3.0,5.2,2.3,Iris-virginica
146,6.3,2.5,5.0,1.9,Iris-virginica
147,6.5,3.0,5.2,2.0,Iris-virginica
148,6.2,3.4,5.4,2.3,Iris-virginica
149,5.9,3.0,5.1,1.8,Iris-virginica


In [8]:
df.shape

(150, 5)

In [9]:
df.columns

Index(['SepalLengthCm', 'SepalWidthCm', 'PetalLengthCm', 'PetalWidthCm',
       'Species'],
      dtype='str')

In [10]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   SepalLengthCm  150 non-null    float64
 1   SepalWidthCm   150 non-null    float64
 2   PetalLengthCm  150 non-null    float64
 3   PetalWidthCm   150 non-null    float64
 4   Species        150 non-null    str    
dtypes: float64(4), str(1)
memory usage: 7.9 KB


In [11]:
df.isnull().sum()

SepalLengthCm    0
SepalWidthCm     0
PetalLengthCm    0
PetalWidthCm     0
Species          0
dtype: int64

In [12]:
print("Dataset loaded successfully!")
print("Number of records:", len(df))
print("Number of columns:", len(df.columns))
print("Missing values:", df.isnull().sum().sum())

Dataset loaded successfully!
Number of records: 150
Number of columns: 5
Missing values: 0


## Dataset Source

Dataset: Iris Dataset

Source: UCI Machine Learning Repository

https://archive.ics.uci.edu/dataset/53/iris

The Iris dataset contains 150 instances and 4 features, with 3 classes of Iris flowers.

## Conclusion

The Iris CSV dataset was successfully loaded using Pandas. The first and last records were displayed, and the dataset was verified to contain 150 records and 5 columns with no missing values.